### PR TITLE
Add design document

### PR DESCRIPTION
**Pull request by SchSeba**
Add design document into the repo.

### CODE REVIEW

File: doc/design.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      
+# Post admission control intervention:
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod0
+  annotations:
+    k8s.v1.cni.cncf.io/networks: '[
+       { "name" : "macvlan-conf-3","mac": "c2:b0:57:49:47:f1" }
+      ]
```
Comment (phoracek) [inline doc/design.md]: missing `'`

File: doc/design.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+This document proposes a collision-free allocation of MAC addresses within a single cluster.
+
+### Allocation
+This specification will describe an intermediate step prior to the initialization of Pods within a kubernetes cluster. During this step, a Pod's template will be read, processed, and modified to specify a MAC address.
+
+It is also to be noted that besides Pods, the proposed behavior will be generic enough to support future entities that may require MAC address allocation. For convenience sake, I will only refer to Pods throughout this document.
+
+### Collision prevention
+This specification will describe how the allocation process (as described in 3.1) will ensure no collisions may occur.
```
Comment (phoracek) [inline doc/design.md]: there is no "3.1", please mention the section by its name

File: doc/design.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+## Design
+Both the act of modifying a Pod's template and deriving available addresses will be done by a new designated admission controller. The admission controller will operate within a predefined address range. A user will be able to pass his own desired range, or let a default range take place.
```
Comment (phoracek) [inline doc/design.md]: s/his/their/

File: doc/design.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+### Allocation
+In order to control the allocation of MAC addresses for network devices, the admission controller will intercept the initialization of a Pod and observe its NetworkAttachmentDefinition's "mac" field.
+
+There are 2 possibilities for the "mac" field:
+
+It is unpopulated. In such case, the CNI assigns a MAC address.
+
+A static MAC address is present. In such case, the CNI assigns the specified address.
```
Comment (phoracek) [inline doc/design.md]: Mention whether we check that desired address is already allocated

File: doc/design.md (hunk 2 above)
Comment (SchSeba) [reply doc/design.md]: done

File: doc/design.md (hunk 3 above)
Comment (SchSeba) [reply doc/design.md]: done

File: doc/design.md (hunk 1 above)
Comment (SchSeba) [reply doc/design.md]: done

File: doc/design.md (hunk 4 above)
Comment (SchSeba) [reply doc/design.md]: done

**Pull request merged**